Content.Server/Engineering/EntitySystems/SpawnAfterInteractSystem.cs

```diff
@@ -65,7 +65,7 @@ bool IsTileClear()
 
             EntityManager.SpawnEntity(component.Prototype, args.ClickLocation.SnapToGrid(grid));
 
-            if (component.RemoveOnInteract && stackComp == null && !((!EntityManager.EntityExists(uid) ? EntityLifeStage.Deleted : EntityManager.GetComponent<MetaDataComponent>(component.Owner).EntityLifeStage) >= EntityLifeStage.Deleted))
+            if (component.RemoveOnInteract && stackComp == null && !((!EntityManager.EntityExists(uid) ? EntityLifeStage.Deleted : EntityManager.GetComponent<MetaDataComponent>(uid).EntityLifeStage) >= EntityLifeStage.Deleted))
```

Comment:
make this just use TryDel if its in current engine, and if not then use `... && !Deleted(uid)) Del(uid);`

Reply:
the condition is such a clusterfuck, I hope this is what you meant